.travis/tooling.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+
+from __future__ import print_function
+
+import os
+import subprocess
+
+
+def changed_files(*args):
+    """
+    Returns a set of changed files in a given commit range.
+
+    :param commit_range: Arguments to pass to ``git diff``.
+    """
+    files = set()
+    command = ['git', 'diff', '--name-only'] + list(args)
+    diff_output = subprocess.check_output(command).decode('ascii')
+    for line in diff_output.splitlines():
+        filepath = line.strip()
+        if filepath:
+            files.add(filepath)
+    return files
+
+
+def are_there_job_relevant_changes(changed_files, task):
+    """
+    Given a list of changed files and a Make task, are there any reasons
+    why the output of this task might have changed?
+
+    e.g. if the loris/ directory changes, we should re-run tasks associated
+         with Loris.
+
+    Returns a list of reasons why the task should be run.
+    """
+    reasons = []
+
+    if task == 'check-format':
```

Comment:
lint-ontologies?

Reply:
Turns out it’s subsumed by the `check-format` task, so we're fine.